lib/azure/armrest/model/base_model.rb

```diff
@@ -22,6 +22,8 @@ def self.attr_hash(*attrs)
 
       attr_hash :tags
 
+      attr_accessor :response_headers
```

Comment:
Similarly add response_headers as an optional hash argument in the initializer.
How useful are the headers? The model object if returned by a single GET has headers, but if in a list does not have headers. Will the difference cause trouble for application?